doc/Requirements.md

```diff
@@ -0,0 +1,526 @@
+(Why: A parameter that can not be accessed is of no use.)
+
+A value MUST be convertible to a string.
+(Why: To display a parameter value to the human user for debugging purpose.)
+
+A bijective conversion string <-> value SHOULD exist.
+(Why: To support serialization and deserialization of the parameter values.
+      It is not a MUST because a PF implementation could offer serialization to
+      custom -not string based- format.)
+
+### Data type
+
+#### Definition
+All parameters have a data type.
+A data type designates parameter invariants.
+A data type is the the meaning of the data and the way values of that type can be stored.
+
+
+#### Philosophy
+
+A data type defines the value properties:
+ - memory layout
+ - value constrains
+
+A value type is mostly used to:
+ - pretty display parameter values (not just a as an array of bits)
+ - check for user error when setting it (out of bound, invalid...)
+ - offer a type safe api
+
+#### Requirements
+
+A PF SHOULD support the following types.
+If a type is chosen to be supported, it MUST respect all MUST clause,
+SHOULD respect all SHOULD clause, MAY respect all MAY clause of the type.
+(Why: All type are not necessary to use the PF. For example any parameter could
+      be represented as an array of char (string). But this would not permit to
+      check parameter validity (invariants) as well a pretty displaying the values.)
+
+Implementation MAY add another API to access a parameter value.
+(Why: For example a C++ implementation may give access to a string as an std::string object.)
+
+##### Integers
+PF SHOULD support signed and unsigned integer parameters
+(Why: The reference implementation supports it.)
+
+PF MUST support integer with invariant size.
+(Why: It is common in C API to expect numbers to have a fixed maximum size.)
+
+The API to access it MUST respect C integer abi.
+(Why: For easy access from C code.)
+
+Supported integer size SHOULD be at least 8, 16 and 32 bits.
+(Why: The reference implementation supports it.)
+
+PF MAY support constraining the parameter minimum and maximum value.
+(Why: To catch user out of valid range errors when changing the parameter value.)
+
+##### String
+PF SHOULD support array of characters.
+(Why: Everything that a computer can store fits in an array of characters.
+      It can be used as a fallback type if no other matches the parameter.)
+
+The array maximum size MAY be invariant (immutable).
+(Note: This is what the reference implementation does.)
+
+The API to access the string value SHOULD support null terminated character array.
+As it is commonly done in C.
+(Why: For easy access from C code.)
+
+##### Fix point parameter
+PF SHOULD support fix point parameters. Ie integers divided by a fixed power of two.
+(Why: The reference implementation supports it.)
+
+The API to access the values SHOULD respect the Qm.n and UQm.n standards.
+(Why: It is the main standard for fix point parameters.)
+
+PF SHOULD support at least m + n ∈ [0,32]
```

Comment:
- This does not render well because of the brackets
- This is not true for Signed Qn.m

Replace with: `0 <= m + n <= 31` for a Signed Qn.m and `0 <= m + n <= 32` for an Unsigned Qn.m.